### PR TITLE
Fix #16

### LINKED ISSUE #16: Omnicompletion for the new native lsp completion
When I set up neovim using https://neovim.io/doc/user/lsp.html#vim.lsp.omnifunc(), this plugin fails with the following error

Vim(return):E117: Unknown function: v:lua.vim.lsp.omnifunc

How can I fix this?

### PR DESCRIPTION
**Pull request by acomagu**
Fix crashing if the `omnifunc` is `v:lua.vim.lsp.omnifunc`.

I hope anyone suggests more smart solution/code! I'm not good at VimScript. :pray:

Fix #16

### CODE REVIEW

File: autoload/asyncomplete/sources/omni.vim [hunk 1]
```diff
@@ -34,12 +34,20 @@ endfunction
...
+      return s:call_nvim_lsp_omnifunc(a:1, a:2)
+    elseif
+      return call(&omnifunc, a:000)
+    endif
   finally
     call setpos('.', cursor)
   endtry
 endfunction
 
+function! s:call_nvim_lsp_omnifunc(a:fundstart, a:base) abort
```
Comment (entropitor) [inline autoload/asyncomplete/sources/omni.vim]: typo? 
```suggestion
function! s:call_nvim_lsp_omnifunc(a:findstart, a:base) abort
```

**Comment by entropitor** [comment]
@acomagu Something like the below seems to be "working" for me
```vim
function! s:safe_omnifunc(...) abort
  let cursor = getpos('.')
  try
    if &omnifunc == 'v:lua.vim.lsp.omnifunc'
      return v:lua.vim.lsp.omnifunc(a:1, a:2)
    elseif
      return call(&omnifunc, a:000)
    endif
  finally
    call setpos('.', cursor)
  endtry
endfunction
```

At least it seems to be logging the correct results in my asyncomplete log but somehow it's not showing the popup menu, is that working for you?

**Comment by entropitor** [comment]
I had to also revert https://github.com/yami-beta/asyncomplete-omni.vim/pull/3/files to be able to see the popup menu with the builtin lsp

**Comment by acomagu** [comment]
Yes it seems to show popup correctly for me like:
outout

Without asyncomplete-omni.vim, I need to type \<C-x>\<C-o> to show that.

For me, it also works with reverting #3 as well so maybe it should be included?(I don't understand the meaning of #3 changes).

**Comment by acomagu** [comment]
I think we need @yami-beta 's opinion.

**Comment by entropitor** [comment]
I also noticed it wasn't working very well as it wasn't getting the correct context, so I'm currently using this fork (part of it was copied out of the lsp provider for asyncomplete): https://github.com/yami-beta/asyncomplete-omni.vim/compare/master...entropitor:master

**Comment by yami-beta** [comment]
I agree to merge this Pull Request.
Can I merge this?

**Comment by acomagu** [comment]
@yami-beta Can you decide whether to include the diff which reverting #3(and some more?)? @entropitor says it works only if revert #3, as [master...entropitor:master](https://github.com/yami-beta/asyncomplete-omni.vim/compare/master...entropitor:master).
In my environment, it seems to work correctly without it. And I could not understand the meaning of #3 so I could not judge that.

**Comment by entropitor** [comment]
For me this is only part of the fix

I needed:
1. This
2. Revert of #3
3. Some changes to how to get the text sent to omnicomplete. I copied this from asyncomplete lsp, see my fork. Otherwise it was sending the whole line and I got garbage completions (all the variables in scope, including globals, no matter what I typed) all the time and it would not start filtering them as every character it would refresh the list and get the same garbage out, no matter what I typed. At least for typescript.

**Comment by yami-beta** [comment]
I agree to merge this Pull Request with revert #3.
Thank you !

**Pull request merged**